### DISCUSSION TITLE
Create a dbus-cxx recipe for yocto

### DISCUSSION THREAD

**Discussion by caiotoledo** [dbus-cxx/dbus-cxx#69] [Ideas]
Some of the wrappers for dbus has its own recipes in yocto to be used in embedded Linux projects.

I have a side project that I wanted to use your library and I created my own recipe to build your library.

Do you mind if I share this recipe to the OpenEmbedded?
Some wrappers like dbus-cpp and dbus-glib already have its own recipe in their layers.

**Comment by rm5248** [comment]
That would be great if you could do that.  Just be aware that the (current) release probably doesn't work well with libglib, so you may not want to create a recipe that builds the glib support.

**Comment by caiotoledo** [comment]
Thanks for the answer!

What release do you recommend? I was using 0.12.0.

**Reply by rm5248** [reply]
That is the latest; I'm close to releasing a 2.0 version hopefully in the next few weeks.  If you need something stable, stick with 0.12.0.

**Comment by caiotoledo** [comment]
@rm5248,

I've created a PR for the yocto recipe -> https://github.com/openembedded/meta-openembedded/pull/302

Seems that was already merge in:
- https://github.com/openembedded/meta-openembedded/commit/7282a97430a1ea24e516ad95b6a36c5751b9d256
- https://github.com/openembedded/meta-openembedded/commit/a6373a21eaa7e26478c770dc02462149a8d895ec
- https://github.com/openembedded/meta-openembedded/commit/556293d49f973db863c82b211036f22171ffa7e6

It was pretty straight forward, I had only a problem to compile with `musl` but I've created a simple patch (https://github.com/openembedded/meta-openembedded/commit/556293d49f973db863c82b211036f22171ffa7e6) to solve it.
Do you pretend to give support to `musl`? If yes, I can create a PR with this patch in dbus-cxx if you wish, otherwise I can keep as a patch in the recipe.

**Reply by rm5248** [reply]
> Seems that was already merge in:
> 
>     * [openembedded/meta-openembedded@7282a97](https://github.com/openembedded/meta-openembedded/commit/7282a97430a1ea24e516ad95b6a36c5751b9d256)

You may not want to depend on master for the recipe.  I think it's fine since it has a SRCREV, but the master branch will contain the new 2.0 version shortly, so if it always gets the latest version things may break.  Note that the 2.0 version requires sigc++-3.0, but looking through some of the yocto files it seems like that already exists, so that's good.

> It was pretty straight forward, I had only a problem to compile with `musl` but I've created a simple patch ([openembedded/meta-openembedded@556293d](https://github.com/openembedded/meta-openembedded/commit/556293d49f973db863c82b211036f22171ffa7e6)) to solve it.
> Do you pretend to give support to `musl`? If yes, I can create a PR with this patch in dbus-cxx if you wish, otherwise I can keep as a patch in the recipe.

I don't see an issue with that.  If you are able to test with the dbus-cxx-2 branch with musl I would really appreciate it; I'm only going to be doing bugfixes from now on with the 0.x.y series of releases.

**Reply by caiotoledo** [reply]
> You may not want to depend on master for the recipe. I think it's fine since it has a SRCREV, but the master branch will contain the new 2.0 version shortly, so if it always gets the latest version things may break.

In this case it's not a problem to depend on the master branch, as I have a fixed `SRCREV` with the hash from 0.12.0 tag, that's the usual approach for yocto recipes. But it's possible to depend on the git tag instead of `SRCREV`.

> Note that the 2.0 version requires sigc++-3.0, but looking through some of the yocto files it seems like that already exists, so that's good.

Nice! When you have an official 2.0 release I can update the recipe quickly, just by changing its depedencies and `SRCREV`.

> I don't see an issue with that. If you are able to test with the dbus-cxx-2 branch with musl I would really appreciate it; I'm only going to be doing bugfixes from now on with the 0.x.y series of releases.

I have a patch that's applied during bitbake do_patching process -> https://github.com/openembedded/meta-openembedded/blob/master/meta-oe/recipes-core/dbus-cxx/files/fix_build_musl.patch
I can try to check if it works and then open a Pull request for `dbus-cxx-2`.
